app/src/main/java/org/simple/clinic/patient/PatientSearchResultDataSource.kt

```diff
@@ -55,11 +55,27 @@ class PatientSearchResultDataSource(
     val loadParamsForDatabaseSource = LoadRangeParams(params.startPosition, params.loadSize)
     source.loadRange(loadParamsForDatabaseSource, object : LoadRangeCallback<PatientSearchResult>() {
```

Comment:
We cannot call this with the params directly. This is because we have added additional items to the final set of items (the facility header, etc) which are unknown to the data source created for Room. We will need to account for these additional items when passing a set of params to load the next page.